dev/system-test/firestore.ts

```diff
@@ -1018,6 +1022,30 @@ describe('DocumentReference class', () => {
     return promise;
   });
 
+  it.only('can write and read vector embeddings', async () => {
+    const ref = randomCol.doc();
+    await ref.create({
+      vectorEmpty: FieldValue.vector(),
+      vector1: FieldValue.vector([1, 2, 3.99]),
+    });
+    await ref.set({
+      vectorEmpty: FieldValue.vector(),
+      vector1: FieldValue.vector([1, 2, 3.99]),
+      vector2: FieldValue.vector([0, 0, 0]),
+    });
+    await ref.update({
+      vector3: FieldValue.vector([-1, -200, -999]),
+    });
+
+    const snap1 = await ref.get();
+    expect(snap1.get('vectorEmpty')).to.deep.equal(FieldValue.vector());
+    expect(snap1.get('vector1')).to.deep.equal(FieldValue.vector([1, 2, 3.99]));
```

Comment:
Consider using `VectorValue.isEqual` instead of `to.deep.equal` because `to.deep.equal` relies on equality testing using private members.

Reply:
I want to change this in the other PR to save myself from conflict resolving, hopefully that is OK with you.